### PR TITLE
[clang-tidy] readability-simplify-boolean-expr

### PR DESCRIPTION
**Pull request by jwhui**

### CODE REVIEW

**Review by wgtdkp — APPROVED** [review]
LGTM :+1:

File: src/ncp/ncp_base_mtd.cpp [hunk 1]
```diff
@@ -679,7 +679,7 @@ template <> otError NcpBase::HandlePropertySet<SPINEL_PROP_THREAD_ALLOW_LOCAL_NE
     SuccessOrExit(error = mDecoder.ReadBool(value));
 
     // Register any net data changes on transition from `true` to `false`.
-    shouldRegisterWithLeader = (mAllowLocalNetworkDataChange == true) && (value == false);
+    shouldRegisterWithLeader = (mAllowLocalNetworkDataChange) && (!value);
```
Comment (simonlingoogle) [inline src/ncp/ncp_base_mtd.cpp:682]:
```suggestion
    shouldRegisterWithLeader = mAllowLocalNetworkDataChange && !value;
```
nit: remove redundant () ?

File: src/ncp/ncp_base_mtd.cpp [hunk 2]
```diff
@@ -819,7 +819,7 @@ template <> otError NcpBase::HandlePropertySet<SPINEL_PROP_SERVER_ALLOW_LOCAL_DA
     SuccessOrExit(error = mDecoder.ReadBool(value));
 
     // Register any server data changes on transition from `true` to `false`.
-    shouldRegisterWithLeader = (mAllowLocalServerDataChange == true) && (value == false);
+    shouldRegisterWithLeader = (mAllowLocalServerDataChange) && (!value);
```
Comment (simonlingoogle) [inline src/ncp/ncp_base_mtd.cpp:822]: ditto

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1: 
with nits

**Review by abtink — APPROVED** [review]
LGTM 👍

File: src/ncp/ncp_base_mtd.cpp [hunk 3]
```diff
@@ -277,9 +277,9 @@ template <> otError NcpBase::HandlePropertySet<SPINEL_PROP_NET_STACK_UP>(void)
     SuccessOrExit(error = mDecoder.ReadBool(enabled));
 
     // If the value has changed...
-    if ((enabled != false) != (otThreadGetDeviceRole(mInstance) != OT_DEVICE_ROLE_DISABLED))
+    if ((enabled) != (otThreadGetDeviceRole(mInstance) != OT_DEVICE_ROLE_DISABLED))
```
Comment (abtink) [inline src/ncp/ncp_base_mtd.cpp:280]:
```suggestion
    if (enabled != (otThreadGetDeviceRole(mInstance) != OT_DEVICE_ROLE_DISABLED))
```

**Pull request merged**